runtime/src/registry/mod.rs

```diff
@@ -0,0 +1,264 @@
+        RegistryDoesNotExist,
+        /// The registry id is too large.
+        RegistryOverflow,
+        /// Unable to recreate the anchor hash from the proofs and data provided. This means
+        /// the [validate_proofs] method failed.
+        InvalidProofs,
+        /// The values vector provided to a mint call doesn't match the length of the specified
+        /// registry's fields vector.
+        InvalidMintingValues,
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        type Error = Error<T>;
+        fn deposit_event() = default;
+
+        #[weight = 10_000]
+        pub fn create_registry(origin,
+                               info: RegistryInfo,
+        ) -> dispatch::DispatchResult {
+            ensure_signed(origin)?;
+
+            let registry_id = <Self as VerifierRegistry>::create_registry(info)?;
+
+            // Emit event
+            Self::deposit_event(Event::<T>::RegistryCreated(registry_id));
+
+            Ok(())
+        }
+
+        #[weight = 10_000]
+        pub fn mint(origin,
+                    owner_account: <T as frame_system::Trait>::AccountId,
+                    registry_id: RegistryId,
+                    token_id: TokenId,
+                    asset_info: T::AssetInfo,
+                    mint_info: MintInfo<<T as frame_system::Trait>::Hash, H256>,
+        ) -> dispatch::DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            // Internal mint validates proofs and modifies state or returns error
+            let asset_id = AssetId(registry_id, token_id);
+            <Self as VerifierRegistry>::mint(&who,
+                                             &owner_account,
+                                             &asset_id,
+                                             asset_info,
+                                             mint_info)?;
+
+            // Mint event
+            Self::deposit_event(RawEvent::Mint(registry_id, token_id));
+
+            Ok(())
+        }
+    }
+}
+
+// Auxillary methods of the module for internal use
+impl<T: Trait> Module<T> {
+    fn get_document_root(anchor_id: T::Hash) -> Result<H256, dispatch::DispatchError> {
+        let root = match <anchor::Module<T>>::get_anchor_by_id(anchor_id) {
+            Some(anchor_data) => Ok(anchor_data.doc_root),
+            None => Err(Error::<T>::DocumentNotAnchored),
+        }?;
+
+        Ok( H256::from_slice(root.as_ref()) )
+    }
+
+    fn create_new_registry_id() -> Result<RegistryId, dispatch::DispatchError> {
+        let id = <RegistryNonce>::get();
+
+        // TODO: Make a U160 type for RegistryId with the uint crate.
+        // Passing through U256 is inefficient and H160 is unneeded.
+        let mut res = Vec::<u8>::with_capacity(32);
+        unsafe { res.set_len(32); }
+        // U256 > H160 so no need for a checked_add
+        U256::from_big_endian(id.as_bytes())
```

Comment:
So the registry id would be formed as:
```
id = [1,0,0,0,0,0,...]
nplus1 = U256(id).add(1) = [2,0,0,0,0,...]
```
so it would be a byte array nonce, right?

Reply:
That's correct. The RegistryId is still an H160. I wanted to make a U160 type for efficiency and ease of use, but had version conflicts with Substrate when importing the shared crate, uint. May take a while to resolve and probably isn't critical.